src/main/res/values/strings.xml

```diff
@@ -139,23 +139,23 @@
   <string name="unknown_error_description">Unknown error.</string>
   <string name="unknown_format">Unknown: %s</string>
   <string name="unsigned">Unsigned</string>
-  <string name="unsigned_description">Unsigned. The list of applications could not be verified. Be careful with
-    applications downloaded from unsigned repositories.</string>
+  <string name="unsigned_description">Unsigned. Could not verify the application list. Be careful downloading
+    applications from unsigned repositories.</string>
   <string name="unstable_updates">Unstable updates</string>
-  <string name="unstable_updates_summary">Suggest updates to unstable versions</string>
+  <string name="unstable_updates_summary">Suggest installing unstable versions</string>
   <string name="unverified">Unverified</string>
   <string name="update">Update</string>
-  <string name="update_notifications">Update notifications</string>
-  <string name="update_notifications_summary">Show a notification when updates are available</string>
+  <string name="update_notifications">Notifications about newer applications</string>
+  <string name="update_notifications_summary">Show a notification when new versions are available</string>
   <string name="updates">Updates</string>
   <string name="updates_empty_summary">All applications up to date</string>
   <string name="username">Username</string>
-  <string name="username_is_not_specified">Username is not specified</string>
-  <string name="validation_index_error_description">Failed to validate the index.</string>
-  <string name="validation_package_error_description_format">Failed to validate the package: %s.</string>
-  <string name="validation_error_hash_lower">invalid hash</string>
-  <string name="validation_error_integrity_lower">integrity check failed</string>
-  <string name="validation_error_permissions_lower">missing permissions</string>
+  <string name="username_is_not_specified">Username missing</string>
+  <string name="validation_index_error_description">Could not validate the index.</string>
+  <string name="validation_package_error_description_format">Could not validate the package: %s.</string>
+  <string name="validation_error_hash_lower">invalid checksum</string>
+  <string name="validation_error_integrity_lower">could not check integrity</string>
+  <string name="validation_error_permissions_lower">add required permissions</string>
```

Comment:
"Could not validate the package: add required permissions" is wrong. Validation failed due to missing permission in metadata (meaning metadata is invalid, or "wrong" APK is downloaded due to repository error or some sort of compromentation).

Reply:
So the package manager only has access to install one metadata signature, and downloading the wrong package faults this?

Reply:
It's sanity check on my side. Since Android 6.0 package installer doesn't show permissions anymore (only "important" ones). So Foxy Droid (as well as the official F-Droid client) displays all permissions on the screen before user press "install". This check is used to ensure user was aware of all permissions required by application. So if this check fails, it shows a notification with "Could not validate the package: missing permissions" error message.